src/libraries/Common/src/System/Security/Cryptography/MLDsaCng.Windows.cs

```diff
@@ -246,51 +274,92 @@ protected override unsafe bool VerifyDataCore(ReadOnlySpan<byte> data, ReadOnlyS
             }
         }
 
-        internal static MLDsaCng ImportPkcs8PrivateKey(ReadOnlySpan<byte> source, out int bytesRead)
+        // TODO why can't we put this attribute on the class directly?
+        [SupportedOSPlatform("windows")]
+        internal static MLDsaCng ImportPkcs8PrivateKey(byte[] source, out int bytesRead)
         {
             int len;
 
             try
             {
-                AsnDecoder.ReadEncodedValue(
-                    source,
-                    AsnEncodingRules.BER,
-                    out _,
-                    out _,
-                    out len);
+                AsnDecoder.ReadEncodedValue(source, AsnEncodingRules.DER, out _, out _, out len);
```

Comment:
Why did you change BER to DER here? (And why did you reformat the code?)

SPKI is DER, because it comes from X.509 (which is all DER).  PKCS#8 comes from PKCS#8 (ha ha ha), aka RFC 5208, which specifies the less restrictive BER encoding.